### PR TITLE
remove '__cdecl' from the demangled name (and not only '__thiscall')

### PR DESCRIPTION
**Pull request by bellenot**

### CODE REVIEW

**Comment by bellenot** [comment]
@phsft-bot build again

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -5848,8 +5848,13 @@ void* TCling::LazyFunctionCreatorAutoload(const std::string& mangled_name) {
    //  function name.
    //
 
-   if (!strncmp(name.c_str(), "public: __thiscall ", sizeof("public: __thiscall ")-1)) {
-      name.erase(0, sizeof("public: __thiscall ")-1);
+   std::string::size_type pos = name.find("__thiscall ");
```
Comment (Axel-Naumann) [inline core/metacling/src/TCling.cxx]: Can we transform this into
`for (auto thingToRemove: {"__thiscall ", "__cdecl ", "typeinfo for ", "vtable for "}) {`
?

**Comment by bellenot** [comment]
@phsft-bot build just on slc6/gcc49

**Comment by bellenot** [comment]
@phsft-bot build

**Comment by bellenot** [comment]
@phsft-bot build again, please

**Pull request merged**